tests/callbacks/test_quantization.py

```diff
@@ -139,3 +154,91 @@ def test_quantization_triggers(tmpdir, trigger_fn: Union[None, int, Callable], e
+def test_quantization_disable_observers(tmpdir, observer_enabled_stages):
+    """Test disabling observers"""
+    qmodel = RegressionModel()
+    qcb = QuantizationAwareTraining(observer_enabled_stages=observer_enabled_stages)
+    trainer = Trainer(callbacks=[qcb], default_root_dir=tmpdir)
+
+    # Quantize qmodel.
+    qcb.on_fit_start(trainer, qmodel)
+    fake_quants = list(module for module in qmodel.modules() if isinstance(module, FakeQuantizeBase))
+    # Disable some of observers before fitting.
+    for fake_quant in fake_quants[::2]:
+        fake_quant.disable_observer()
+
+    for stage, on_stage_start, on_stage_end in [
+        ("train", qcb.on_train_start, qcb.on_train_end),
+        ("validate", qcb.on_validation_start, qcb.on_validation_end),
+        ("test", qcb.on_test_start, qcb.on_test_end),
+        ("predict", qcb.on_predict_start, qcb.on_predict_end),
+    ]:
+        before_stage_observer_enabled = torch.as_tensor(list(map(_get_observer_enabled, fake_quants)))
+
+        on_stage_start(trainer, qmodel)
+        expected_stage_observer_enabled = torch.as_tensor(
+            before_stage_observer_enabled if stage in observer_enabled_stages else [False] * len(fake_quants)
+        )
+        assert torch.equal(
+            torch.as_tensor(list(map(_get_observer_enabled, fake_quants))), expected_stage_observer_enabled
+        )
+
+        on_stage_end(trainer, qmodel)
+        assert torch.equal(
+            torch.as_tensor(list(map(_get_observer_enabled, fake_quants))), before_stage_observer_enabled
+        )
+
+
+@RunIf(quantization=True)
+def test_quantization_val_test_predict(tmpdir):
+    """Test the default quantization aware training not affected by validating, testing and predicting"""
+    seed_everything(42)
+    num_features = 16
+    dm = RegressDataModule(num_features=num_features)
+    qmodel = RegressionModel()
+
+    val_test_predict_qmodel = copy.deepcopy(qmodel)
+    trainer = Trainer(
+        callbacks=[QuantizationAwareTraining(quantize_on_fit_end=False)],
+        default_root_dir=tmpdir,
+        limit_train_batches=1,
+        limit_val_batches=1,
+        limit_test_batches=1,
+        limit_predict_batches=1,
+        val_check_interval=1,
+        num_sanity_val_steps=1,
+        max_epochs=4,
+    )
+    trainer.fit(val_test_predict_qmodel, datamodule=dm)
+    trainer.validate(model=val_test_predict_qmodel, verbose=False)
+    trainer.test(model=val_test_predict_qmodel, verbose=False)
+    trainer.predict(
+        model=val_test_predict_qmodel, dataloaders=[torch.utils.data.DataLoader(RandomDataset(num_features, 16))]
+    )
+
+    expected_qmodel = copy.deepcopy(qmodel)
+    # No validation in ``expected_qmodel`` fitting.
+    Trainer(
+        callbacks=[QuantizationAwareTraining(quantize_on_fit_end=False)],
+        default_root_dir=tmpdir,
+        limit_train_batches=1,
+        limit_val_batches=0,
+        max_epochs=4,
+    ).fit(expected_qmodel, datamodule=dm)
```

Comment:
validation is part of the fit

Reply:
It seems that the validation and the sanity check will be disabled when `limit_val_batches=0` is passed.

Reply:
then rather disable validation dataloader?

Reply:
Does that mean
```python
Trainer(
    callbacks=[QuantizationAwareTraining(quantize_on_fit_end=False)],
    default_root_dir=tmpdir,
    limit_train_batches=1,
    max_epochs=4,
).fit(expected_qmodel, train_dataloaders=dm.train_dataloader())
```

Reply:
Yes, limit_val_batches=0 deactivates both. To disable sanity checking, just need `Trainer(num_sanity_val_steps=0)`.